active/NavigationView/HierarchicalNav/HierarchicalNavigationView.md

```diff
@@ -0,0 +1,265 @@
+        {
+            var nvib = e.ExpandingItemContainer;
+            if(nvib != null)
+            {
+                var name = "Last Expanding: " + nvib.Content;
+                ExpandingItemLabel.Text = name;
+            }
+            else
+            {
+                ExpandingItemLabel.Text = "Last Expanding: ERROR - No container returned!";
+            }
+        }
+
+        private void CollapsedItem(object sender, NavigationViewCollapsedEventArgs e)
+        {
+            var nvib = e.CollapsedItemContainer;
+            if (nvib != null)
+            {
+                var name = "Last Collapsed: " + nvib.Content;
+                CollapsedItemLabel.Text = name;
+            }
+            else
+            {
+                CollapsedItemLabel.Text = "Last Collapsed: ERROR - No container returned!";
+            }
+        }
+```
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples
+section.  For example, see the Remarks for the MediaPlayerElement 
+(https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.MediaPlayerElement#remarks). -->
+
+
+# API Notes
+<!-- Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. -->
+
+
+# API Details
+```c++
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewExpandingEventArgs
+{
+    NavigationViewItemBase ExpandingItemContainer { get; };
+}
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewCollapsedEventArgs
+{
+    NavigationViewItemBase CollapsedItemContainer { get; };
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationView", f209ce15-391a-42ca-9fc6-f79da65aca32)]
+[WUXC_STATIC_NAME("INavigationViewStatics", 363a86c7-72da-4420-b871-15d9d0d45756)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewFactory", e50687c1-b7c2-4975-ad7a-5f4fe6a514c9)]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
+unsealed runtimeclass NavigationView : Windows.UI.Xaml.Controls.ContentControl
+{
+...
+    [WUXC_VERSION_PREVIEW]
+    {
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewExpandingEventArgs> Expanding;
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewCollapsedEventArgs> Collapsed;
+
+        void Expand(NavigationViewItem item);
+        void Collapse(NavigationViewItem item);
+    }
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationViewItemBase", edf04eb1-37d1-471f-8570-3829ee5b2bc6)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewItemBaseFactory", eb014cef-7890-4ebb-8245-02e8510f321d)]
+[default_interface]
+unsealed runtimeclass NavigationViewItemBase : Windows.UI.Xaml.Controls.ContentControl
+{
+    [WUXC_VERSION_PREVIEW]
+    {
+        Boolean IsSelected{ get; set; };
```

Comment:
Header and Separator derive from NavigationViewItemBase, but IsSelected might not make sense for those derivations.

Reply:
True. But not adding this property would constitute a breaking change. In the past, NVIB had derived from ListViewItem, which derives from SelectorItem and has the IsSelected property.

Reply:
Ok to keep it since moving it is a breaking change.